server/extension/extension.go

```diff
@@ -624,10 +640,27 @@ func findProxy(registry ProxyRegistry, extName string, dest v1alpha1.Application
 	return nil, fmt.Errorf("no proxy found for extension %q", extName)
 }
 
+// ProxyRegistry returns the proxy registry associated for the given
+// extension name.
+func (m *Manager) ProxyRegistry(name string) (ProxyRegistry, bool) {
+	pReg, found := m.registry[name]
+	return pReg, found
+}
+
 // CallExtension returns a handler func responsible for forwarding requests to the
 // extension service. The request will be sanitized by removing sensitive headers.
-func (m *Manager) CallExtension(extName string, registry ProxyRegistry) func(http.ResponseWriter, *http.Request) {
+func (m *Manager) CallExtension() func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
+		segments := strings.Split(strings.TrimPrefix(r.URL.Path, "/"), "/")
+		if segments[0] != "extensions" {
+			http.Error(w, fmt.Sprintf("Invalid URL: first segment must be %s", URLPrefix), http.StatusBadRequest)
+			return
+		}
+		extName := segments[1]
```

Comment:
Probably should have a length check to avoid runtime errors.

Reply:
Good point. How would you define the length check logic?

Reply:
```go
if len(segments) != 2 {
    return fmt.Errorf("invalid URL: must have two segments, e.g. extensions/my-extension")
}
```

Reply:
Oh.. I see what you mean.. Unfortunately this can't be done. There is no way for us to know how many segments in the URL a given backend would be willing to handle. It could be:
- <argocd-host>/extensions/some-extension/api/do-something
- <argocd-host>/extensions/another-extension/hello/world/everyone

All the code cares about is the second segment to extract the extension name.

Reply:
Then how about 

```go
if len(segments) < 2 {
    return fmt.Errorf("invalid URL: must have at least two segments, e.g. extensions/my-extension")
}
```

Reply:
I added it the code but realized that this is actually impossible because we are registering the http handler adding a `/` at the end which means that length will always be >= 2:
https://github.com/argoproj/argo-cd/pull/15574/files#diff-78f42ba40d0f10b08c73b7e6bb8376f398e249c963cf549e89591d6b6826b9a4R1065

Reply:
Ah cool